torch/csrc/jit/codegen/cuda/graph_fuser.cpp

```diff
@@ -914,19 +916,134 @@ void PeepholeOptimizeShapeExpressions(Block* block) {
   }
 }
 
+// Note: shamelessly copying code from NNC
+// We store given tensor types in `attr::types` at prim::CudaFuserGuard.
+void guardFusionGroup(Node* fusion) {
+  // Fixup types of the subgraph inputs
+  std::vector<TypePtr> guard_types;
+  std::vector<Value*> inputs_to_check;
+  for (Value* input : fusion->inputs()) {
+    // We only check inputs of the fusion group and expect NNC to infer
+    // intermediates and outputs shapes
+    if (!input->type()->cast<TensorType>()) {
+      continue;
+    }
+
+    // fusion outputs are already guarded
+    if (input->node()->kind() == prim::Constant) {
+      // input->node()->kind() == prim::FusionGroup ||
```

Comment:
why commented out? should we remove these lines completely? also, maybe add a comment explaining why?

Reply:
commented